src/OrchardCore/OrchardCore.DisplayManagement.Liquid/Filters/ShapeRenderFilter.cs

```diff
@@ -32,6 +33,10 @@ static async ValueTask<FluidValue> Awaited(Task<IHtmlContent> task)
 
                 return new ValueTask<FluidValue>(new HtmlContentValue(task.Result));
             }
+            else if (obj is IHtmlContent html)
```

Comment:
I want to understand what you are writing that should result in that behavior. It doesn't hurt, but it's kind of wrong to let people use `shape_render` on anything and it just works. Because tomorrow it's a string, which will fail, then an integer ... The dev in me just wants to throw an exception to let them know that a shape should be used for `shape_render`.

But if you show some sample liquid that someone could write and end up to be an `IHtmlContent` passed to `shape_render` I might be more open.

Reply:
The `Form_Wrapper` (on top of this PR discussion) is an exemple of rendering `IHtmlContent`.
The implementation behind is `Microsoft.AspNetCore.Mvc.ViewFeatures.Buffers.ViewBuffer`.

I agree with you and it don't make sense to use `shape_render` on something that is not an `IShape`, but I didn't found an existing suitable filter for `IHtmlContent`. Maybe we need a new dedicated filter ?

Reply:
{{ youHtmlContent }}

Reply:
Now I am having doubts ;)

Reply:
> {{ youHtmlContent }}

It renders the type name

Reply:
We need a converter then. The same kind we do for JObject, to convert an `IHtmlContent` to a `FluidValue`. I am approving this PR but we need to start another that removes this piece of code and provides a way to render an IHtmlContent.

Reply:
One solution would be to make `Shape.Metadata.ChildContent` both an `IHtmlContent` and an `IShape` as the `PositionWrapper` is. But yes, `Shape.Metadata.ChildContent` is not an `IShape`, so looks better to have a filter / converter.

Note: From memory we already have a fluid value that is an `IHtmlContent` => `HtmlContentValue`